src/iface/ethernet.rs

```diff
@@ -826,13 +831,13 @@ impl<'b, 'c> InterfaceInner<'b, 'c> {
         packet
     }
 
-    #[cfg(feature = "proto-ipv4")]
-    fn process_icmpv4<'frame>(&self, _sockets: &mut SocketSet, ip_repr: IpRepr,
-                              ip_payload: &'frame [u8]) -> Result<Packet<'frame>>
+    #[cfg(any(feature = "proto-ipv4", feature = "proto-ipv6"))]
+    fn process_icmp<'frame>(&self, _sockets: &mut SocketSet, ip_repr: IpRepr,
```

Comment:
ICMPv6 packets will be processed by [`process_icmpv6`](https://github.com/m-labs/smoltcp/blob/master/src/iface/ethernet.rs#L741). It has a different [IP protocol number](https://www.iana.org/assignments/protocol-numbers/protocol-numbers.xhtml).